pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -121,10 +93,11 @@ public static boolean isSameReference(Object left, Object right) {
   }
 
   public static int hashCodeOf(Object o) {
```

Comment:
use nullable annotation on the object. Please add javadoc describing what happens if o == null  (intellij will show doc not requiring me to jump to code to understand behavior)